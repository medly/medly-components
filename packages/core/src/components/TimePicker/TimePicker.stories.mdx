import { TimePicker } from './TimePicker';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import { boolean, select, text } from '@storybook/addon-knobs';
import { placements } from '../Popover/Popover.stories.tsx';
import { variants } from './TimePicker.stories';
import { useState } from 'react';

<Meta
    title="Core"
    component={TimePicker}
    parameters={{
        jest: ['TimePicker.test.tsx']
    }}
/>

# TimePicker Component

The `TimePicker` is a controlled component that allows you to select a time through input text or using a dialog. This component is built on top of the `TextField` component. It accepts all the props of `TextField` component. This component will return the time in **24-hour format**.

This component will use the native dialog on the mobile devices.

## Usage

```tsx
import { TimePicker } from '@medly-components/core';

const Component = () => {
    const [time, setTime] = useState('');
    return <TimePicker label="Time" value={time} onChange={setTime} />;
};
```

<Preview withToolbar>
    <Story name="TimePicker">
        {() => {
            const [time, setTime] = useState('');
            return (
                <TimePicker
                    id="dob"
                    value={time}
                    onChange={setTime}
                    disabled={boolean('Disabled', false)}
                    label={text('Label', 'Time')}
                    required={boolean('Required', false)}
                    variant={select('variant', variants, 'outlined')}
                    popoverPlacement={select('Popover placement', placements, 'bottom-start')}
                />
            );
        }}
    </Story>
</Preview>
