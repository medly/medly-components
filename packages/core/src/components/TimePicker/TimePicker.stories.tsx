import { TimePickerProps } from './types';

export const variants: Required<TimePickerProps>['variant'][] = ['outlined', 'filled', 'fusion'];
