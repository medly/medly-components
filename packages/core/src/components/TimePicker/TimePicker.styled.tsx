import { AccessTimeIcon } from '@medly-components/icons';
import styled from 'styled-components';
import Popover from '../Popover';
import { OuterWrapper } from '../TextField/Styled';

export const TimePickerWrapper = styled(Popover)<{ fullWidth?: boolean; minWidth?: string; maxWidth?: string }>`
    display: ${({ fullWidth }) => (fullWidth ? 'flex' : 'inline-flex')};
    width: ${({ fullWidth }) => (fullWidth ? '100%' : 'min-content')};
    min-width: ${({ minWidth }) => minWidth};
    max-width: ${({ maxWidth }) => maxWidth};
    margin: ${({ theme, fullWidth }) =>
        fullWidth ? `${theme.spacing.S2} 0` : `${theme.spacing.S2} ${theme.spacing.S2} ${theme.spacing.S2} 0`};

    & > ${OuterWrapper} {
        margin: 0;
    }
`;

export const TimeIcon = styled(AccessTimeIcon).attrs({ title: 'time-icon' })`
    cursor: pointer;
`;
