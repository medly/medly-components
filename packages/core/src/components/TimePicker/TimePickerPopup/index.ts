export { TimePickerPopup as default } from './TimePickerPopup';
