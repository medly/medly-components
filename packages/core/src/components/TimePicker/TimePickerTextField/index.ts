export { TimePickerTextField as default } from './TimePickerTextField';
