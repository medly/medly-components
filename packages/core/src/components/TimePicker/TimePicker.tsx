import { WithStyle, useCombinedRefs } from '@medly-components/utils';
import type { FC } from 'react';
import { forwardRef, memo, useRef } from 'react';
import { TimePickerWrapper } from './TimePicker.styled';
import TimePickerPopup from './TimePickerPopup';
import TimePickerTextField from './TimePickerTextField';
import { TimePickerProps } from './types';

const Component: FC<TimePickerProps> = memo(
    forwardRef((props, ref) => {
        const isMobile = navigator?.userAgent?.indexOf('Mobi') > -1;
        const inputRef = useCombinedRefs<HTMLInputElement>(ref, useRef(null));
        const id = props.id || props.label?.toLowerCase().replace(/\s/g, '') || 'medly-timepicker';
        const { value, onChange, disabled, className, fullWidth, minWidth, maxWidth, popoverDistance, popoverPlacement, ...restProps } =
            props;

        const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => onChange(e.target.value);

        return (
            <TimePickerWrapper className={className} fullWidth={fullWidth} minWidth={minWidth} maxWidth={maxWidth} interactionType="click">
                <TimePickerTextField id={id} ref={inputRef} disabled={disabled} onChange={handleChange} value={value} {...restProps} />
                {!disabled && !isMobile && (
                    <TimePickerPopup
                        value={value}
                        onChange={onChange}
                        popoverDistance={popoverDistance}
                        popoverPlacement={popoverPlacement}
                    />
                )}
            </TimePickerWrapper>
        );
    })
);
Component.defaultProps = {
    size: 'M',
    label: 'Time',
    variant: 'filled',
    disabled: false,
    required: false,
    fullWidth: false,
    minWidth: '20rem',
    showDecorators: true
};
Component.displayName = 'TimePicker';
export const TimePicker: FC<TimePickerProps> & WithStyle = Object.assign(Component, { Style: TimePickerWrapper });
