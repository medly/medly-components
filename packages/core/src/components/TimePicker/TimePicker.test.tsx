import { fireEvent, render, screen } from '@test-utils';
import { TimePicker } from './TimePicker';

describe('TimePicker', () => {
    beforeAll(() => {
        Object.defineProperty(HTMLElement.prototype, 'clientHeight', { configurable: true, value: 24 });
        // @ts-ignore
        Element.prototype.scrollTo = function ({ top }) {
            this.scrollTop = top;
        };
    });

    afterEach(() => {
        Object.defineProperty(global?.navigator, 'userAgent', { configurable: true, value: { indexOf: () => -1 } });
    });

    it('should render properly', () => {
        const { container } = render(<TimePicker label="Time" value="13:11" onChange={jest.fn()} />);
        fireEvent.click(screen.getByLabelText('Time'));
        expect(container).toMatchSnapshot();
    });

    it('should give time entered in the textfield', () => {
        const mockOnChange = jest.fn();
        render(<TimePicker label="Time" value="13:11" onChange={mockOnChange} />);
        fireEvent.change(screen.getByLabelText('Time'), { target: { value: '22:00' } });
        expect(mockOnChange).toBeCalledWith('22:00');
    });

    it('should give the expected time on selecting time from dialog', () => {
        const mockOnChange = jest.fn();
        render(<TimePicker label="Time" value="" onChange={mockOnChange} />);
        fireEvent.click(screen.getByLabelText('Time'));
        fireEvent.click(screen.getByTitle('hour-arrow-down'));
        fireEvent.click(screen.getByTitle('minutes-arrow-down'));
        fireEvent.click(screen.getByText('PM'));
        fireEvent.click(screen.getByText('Apply'));
        expect(mockOnChange).toBeCalledWith('13:01');
    });

    it('should reset the values on clicking on cancel button', () => {
        const mockOnChange = jest.fn();
        render(<TimePicker label="Time" value="" onChange={mockOnChange} />);
        fireEvent.click(screen.getByLabelText('Time'));
        fireEvent.click(screen.getByTitle('hour-arrow-down'));
        fireEvent.click(screen.getByTitle('minutes-arrow-down'));
        fireEvent.click(screen.getByText('PM'));
        fireEvent.click(screen.getByText('Cancel'));
        expect(mockOnChange).not.toBeCalled();
        expect(screen.queryByText('hour-arrow-down')).not.toBeInTheDocument();
    });

    it('should not render dialog for mobile devices', () => {
        Object.defineProperty(global?.navigator, 'userAgent', { configurable: true, value: { indexOf: () => 1 } });
        render(<TimePicker label="Time" value="" onChange={jest.fn()} />);
        fireEvent.click(screen.getByLabelText('Time'));
        expect(screen.queryByText('hour-arrow-down')).not.toBeInTheDocument();
    });
});
