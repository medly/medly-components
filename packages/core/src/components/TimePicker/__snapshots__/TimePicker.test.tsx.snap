// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker should render properly 1`] = `
.c10 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c10 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c19 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: pointer;
}

.c19 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c12 {
  z-index: 4;
  position: absolute;
  background-color: white;
  top: calc(100% + 0px);
  left: 0%;
}

.c11 {
  pointer-events: none;
  margin-left: 1.6rem;
}

.c11 * {
  fill: #546A7F;
}

.c11:hover * {
  fill: #546A7F;
}

.c11 * {
  fill: #3872D2;
}

.c11:hover * {
  fill: #546A7F;
}

.c8 {
  top: 50%;
  left: 0;
  cursor: text;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: absolute;
  pointer-events: none;
  -webkit-transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  font-size: 1.6rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.6rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  touch-action: manipulation;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  opacity: 1;
  z-index: 1;
  width: 100%;
  max-width: -webkit-min-content;
  max-width: -moz-min-content;
  max-width: min-content;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  height: 5.6rem;
  padding: 0 1.6rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  cursor: text;
  border-radius: 0.4rem 0.4rem 0 0;
  background-color: #eff2f4;
}

.c4 .c7 {
  color: #546A7F;
}

.c4 ~ .sc-avlt8d-0 {
  color: #546A7F;
}

.c4::after {
  content: '';
  width: 100%;
  height: 0;
  position: absolute;
  bottom: 0;
  left: 0;
  box-sizing: border-box;
  border-top: 0.1rem solid #546A7F;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c4:hover::after,
.c4:focus-within::after {
  border-width: 0.2rem;
}

.c4 input {
  box-shadow: 0 0 0 100000px #eff2f4 inset;
}

.c4:focus-within,
.c4:focus-within:hover {
  background-color: #EBF1FA;
}

.c4:focus-within::after,
.c4:focus-within:hover::after {
  border-color: #3872D2;
  border-radius: 0;
}

.c4:focus-within .c7,
.c4:focus-within:hover .c7 {
  color: #3872D2;
}

.c4:focus-within input,
.c4:focus-within:hover input {
  box-shadow: 0 0 0 100000px #EBF1FA inset;
}

.c4:focus-within input::-webkit-input-placeholder,
.c4:focus-within:hover input::-webkit-input-placeholder {
  color: rgba(0,90,238,.2);
}

.c4:focus-within input::-moz-placeholder,
.c4:focus-within:hover input::-moz-placeholder {
  color: rgba(0,90,238,.2);
}

.c4:focus-within input:-ms-input-placeholder,
.c4:focus-within:hover input:-ms-input-placeholder {
  color: rgba(0,90,238,.2);
}

.c4:focus-within input::placeholder,
.c4:focus-within:hover input::placeholder {
  color: rgba(0,90,238,.2);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
}

.c3 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  min-width: 20rem;
  margin: 0.8rem 0;
}

.c6 {
  color: #13181D;
  width: 100%;
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background-color: transparent;
  border: none;
  text-overflow: ellipsis;
  resize: none;
  z-index: 1;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin-bottom: 0.7rem;
}

.c6,
.c6 ~ .sc-1r5xr2a-0 {
  font-size: 1.6rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.6rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
}

.c6.c6.c6:focus {
  outline: none;
  box-shadow: unset;
}

.c6.c6.c6:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c6.c6.c6:focus::-moz-placeholder {
  opacity: 1;
}

.c6.c6.c6:focus:-ms-input-placeholder {
  opacity: 1;
}

.c6.c6.c6:focus::placeholder {
  opacity: 1;
}

.c6:disabled {
  color: #435465;
}

.c6:disabled ~ .c7,
.c6:disabled ~ .sc-1r5xr2a-0 {
  cursor: not-allowed;
}

.c6::-webkit-input-placeholder {
  opacity: 0;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c6::-moz-placeholder {
  opacity: 0;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c6:-ms-input-placeholder {
  opacity: 0;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c6::placeholder {
  opacity: 0;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c6:not(:placeholder-shown) ~ .c7,
.c6:focus ~ .c7 {
  -webkit-transform: translateY(-87%) scale(0.75);
  -ms-transform: translateY(-87%) scale(0.75);
  transform: translateY(-87%) scale(0.75);
}

.c6:not(:placeholder-shown) ~ .sc-1r5xr2a-0 {
  opacity: 1;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  min-width: 20rem;
  margin: 0.8rem 0.8rem 0.8rem 0;
}

.c1 > .c2 {
  margin: 0;
}

.c1 input[type='time']::-webkit-calendar-picker-indicator {
  background: none;
  display: none;
}

.c16 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c28 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 600;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c27 {
  border: none;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  padding: 0.8rem 2.4rem 1rem;
  border-radius: 0.8rem;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c27,
.c27 .c15,
.c27 .c9,
.c27 .c9 * {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c27:hover {
  cursor: pointer;
}

.c27:focus {
  outline: none;
}

.c27:disabled {
  cursor: not-allowed;
}

.c27::after {
  content: '';
  display: block;
  position: absolute;
  width: 0;
  bottom: 0.85rem;
  left: 50%;
  height: 0.15rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0.15rem;
  -webkit-transform: translate(-50%);
  -ms-transform: translate(-50%);
  transform: translate(-50%);
}

.c27:disabled {
  color: #98A7B7;
}

.c27:disabled::after {
  background-color: #98A7B7;
}

.c27:disabled .c9 * {
  fill: #98A7B7;
}

.c27:not(:disabled):not(:hover) {
  color: #3872D2;
}

.c27:not(:disabled):not(:hover)::after {
  background-color: #3872D2;
}

.c27:not(:disabled):not(:hover) .c9 * {
  fill: #3872D2;
}

.c27:not(:disabled):active {
  color: #275093;
}

.c27:not(:disabled):active::after {
  background-color: #275093;
}

.c27:not(:disabled):active .c9 * {
  fill: #275093;
}

.c27:not(:disabled):active::after {
  width: calc(100% - 4.8rem);
}

.c27:not(:disabled):not(:active):hover {
  color: #3061B3;
}

.c27:not(:disabled):not(:active):hover::after {
  background-color: #3061B3;
}

.c27:not(:disabled):not(:active):hover .c9 * {
  fill: #3061B3;
}

.c27:not(:disabled):not(:active):hover::after {
  width: calc(100% - 4.8rem);
}

.c27 .c9 + .c15 {
  margin-left: 0.8rem;
}

.c27 .c15 + .c9 {
  margin-left: 0.8rem;
}

.c29 {
  border: none;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  padding: 0.8rem 2.4rem 1rem;
  border-radius: 0.8rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c29,
.c29 .c15,
.c29 .c9,
.c29 .c9 * {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c29:hover {
  cursor: pointer;
}

.c29:focus {
  outline: none;
}

.c29:disabled {
  cursor: not-allowed;
}

.c29:disabled {
  color: #98A7B7;
  background: #dfe4e9;
}

.c29:disabled .c9 * {
  fill: #98A7B7;
}

.c29:not(:disabled):not(:hover) {
  color: #ffffff;
  background: #3872D2;
}

.c29:not(:disabled):not(:hover) .c9 * {
  fill: #ffffff;
}

.c29:not(:disabled):active {
  color: #ffffff;
  background: #275093;
}

.c29:not(:disabled):active .c9 * {
  fill: #ffffff;
}

.c29:not(:disabled):not(:active):hover {
  color: #ffffff;
  background: #3061B3;
  box-shadow: 0 0.2rem 0.8rem rgba(48,97,179,0.35);
}

.c29:not(:disabled):not(:active):hover .c9 * {
  fill: #ffffff;
}

.c29 .c9 + .c15 {
  margin-left: 0.8rem;
}

.c29 .c15 + .c9 {
  margin-left: 0.8rem;
}

.c13 {
  padding: 1.2rem;
  background: #ffffff;
  box-shadow: 0 0.2rem 0.8rem #b0bcc8;
  border-radius: 0.4rem;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c14 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: center;
  font-size: 1.4rem;
  font-weight: Regular;
  line-height: 2.2rem;
  color: #546A7F;
}

.c14 > *:nth-child(2) {
  -webkit-flex: 0.5;
  -ms-flex: 0.5;
  flex: 0.5;
}

.c17 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 1.8rem 0 2.4rem;
  position: relative;
}

.c17 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c17::before,
.c17::after {
  content: '';
  position: absolute;
  left: -1.2rem;
  width: calc(100% + 2.4rem);
  height: 0.1rem;
  background-color: #C7D0D8;
}

.c17::before {
  top: 3.2rem;
}

.c17::after {
  bottom: 3.2rem;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 2.4rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c20 {
  padding: 0;
  margin: 0;
  height: 2.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  overflow: auto;
  -webkit-scroll-snap-type: y mandatory;
  -moz-scroll-snap-type: y mandatory;
  -ms-scroll-snap-type: y mandatory;
  scroll-snap-type: y mandatory;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  list-style: none;
}

.c20::-webkit-scrollbar {
  display: none;
}

.c20 > li {
  -webkit-scroll-snap-align: center;
  -moz-scroll-snap-align: center;
  -ms-scroll-snap-align: center;
  scroll-snap-align: center;
  font-size: 1.6rem;
  font-weight: Medium;
  line-height: 2.6rem;
  color: #13181D;
}

.c22 {
  position: relative;
  max-height: 2.4rem;
}

.c23 {
  padding: 0;
  margin: 0;
  gap: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  list-style: none;
  -webkit-transition: -webkit-transform 200ms ease;
  -webkit-transition: transform 200ms ease;
  transition: transform 200ms ease;
}

.c24 {
  font-size: 1.4rem;
  font-weight: Regular;
  line-height: 2.2rem;
  color: #546A7F;
  cursor: pointer;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c25 {
  font-size: 1.6rem;
  font-weight: Medium;
  line-height: 2.6rem;
  color: #13181D;
  cursor: pointer;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
}

.c21 {
  -webkit-flex: 0.5;
  -ms-flex: 0.5;
  flex: 0.5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c26 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0 c1"
    id="medly-popover-wrapper"
  >
    <div
      class="c2 c3"
      id="time-input-wrapper"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <input
            aria-describedby="time-helper-text"
            class="c6"
            id="time-input"
            placeholder=" "
            type="time"
            value="13:11"
          />
          <label
            class="c7 c8"
            for="time-input"
          >
            Time
          </label>
        </div>
        <svg
          class="c9 c10 sc-1668uuz-0 c11"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm-.28-13h.06c.4 0 .72.32.72.72v4.54l3.87 2.3c.35.2.46.65.25.99-.2.34-.64.44-.98.24l-4.15-2.49a.99.99 0 0 1-.49-.86V7.72c0-.4.32-.72.72-.72z"
            fill="#000"
            fill-opacity=".54"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      
    </div>
    <div
      class="c12"
      id="medly-popover-popup"
    >
      <div
        class="c13"
      >
        <div
          class="c14"
        >
          <span
            class="c15 c16"
          >
            Hour
          </span>
          <span />
          <span
            class="c15 c16"
          >
            Minutes
          </span>
          <span />
        </div>
        <div
          class="c17"
        >
          <div
            class="c18"
          >
            <svg
              class="c9 c19"
              fill="none"
              height="1em"
              title="hour-arrow-up"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.295 9l-4.59 4.59A.996.996 0 1 0 8.115 15l3.89-3.88 3.88 3.88a.996.996 0 1 0 1.41-1.41L12.705 9a.996.996 0 0 0-1.41 0z"
                fill="#000"
                fill-opacity=".54"
              />
            </svg>
            <ul
              class="c20"
            >
              <li>
                00
              </li>
              <li>
                01
              </li>
              <li>
                02
              </li>
              <li>
                03
              </li>
              <li>
                04
              </li>
              <li>
                05
              </li>
              <li>
                06
              </li>
              <li>
                07
              </li>
              <li>
                08
              </li>
              <li>
                09
              </li>
              <li>
                10
              </li>
              <li>
                11
              </li>
            </ul>
            <svg
              class="c9 c19"
              fill="none"
              height="1em"
              title="hour-arrow-down"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.875 9l-3.88 3.88L8.115 9a.996.996 0 1 0-1.41 1.41l4.59 4.59c.39.39 1.02.39 1.41 0l4.59-4.59a.996.996 0 0 0 0-1.41c-.39-.38-1.03-.39-1.42 0z"
                fill="#000"
                fill-opacity=".54"
              />
            </svg>
          </div>
          <span
            class="c15 c16 c21"
          >
            :
          </span>
          <div
            class="c18"
          >
            <svg
              class="c9 c19"
              fill="none"
              height="1em"
              title="minutes-arrow-up"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.295 9l-4.59 4.59A.996.996 0 1 0 8.115 15l3.89-3.88 3.88 3.88a.996.996 0 1 0 1.41-1.41L12.705 9a.996.996 0 0 0-1.41 0z"
                fill="#000"
                fill-opacity=".54"
              />
            </svg>
            <ul
              class="c20"
            >
              <li>
                00
              </li>
              <li>
                01
              </li>
              <li>
                02
              </li>
              <li>
                03
              </li>
              <li>
                04
              </li>
              <li>
                05
              </li>
              <li>
                06
              </li>
              <li>
                07
              </li>
              <li>
                08
              </li>
              <li>
                09
              </li>
              <li>
                10
              </li>
              <li>
                11
              </li>
              <li>
                12
              </li>
              <li>
                13
              </li>
              <li>
                14
              </li>
              <li>
                15
              </li>
              <li>
                16
              </li>
              <li>
                17
              </li>
              <li>
                18
              </li>
              <li>
                19
              </li>
              <li>
                20
              </li>
              <li>
                21
              </li>
              <li>
                22
              </li>
              <li>
                23
              </li>
              <li>
                24
              </li>
              <li>
                25
              </li>
              <li>
                26
              </li>
              <li>
                27
              </li>
              <li>
                28
              </li>
              <li>
                29
              </li>
              <li>
                30
              </li>
              <li>
                31
              </li>
              <li>
                32
              </li>
              <li>
                33
              </li>
              <li>
                34
              </li>
              <li>
                35
              </li>
              <li>
                36
              </li>
              <li>
                37
              </li>
              <li>
                38
              </li>
              <li>
                39
              </li>
              <li>
                40
              </li>
              <li>
                41
              </li>
              <li>
                42
              </li>
              <li>
                43
              </li>
              <li>
                44
              </li>
              <li>
                45
              </li>
              <li>
                46
              </li>
              <li>
                47
              </li>
              <li>
                48
              </li>
              <li>
                49
              </li>
              <li>
                50
              </li>
              <li>
                51
              </li>
              <li>
                52
              </li>
              <li>
                53
              </li>
              <li>
                54
              </li>
              <li>
                55
              </li>
              <li>
                56
              </li>
              <li>
                57
              </li>
              <li>
                58
              </li>
              <li>
                59
              </li>
            </ul>
            <svg
              class="c9 c19"
              fill="none"
              height="1em"
              title="minutes-arrow-down"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.875 9l-3.88 3.88L8.115 9a.996.996 0 1 0-1.41 1.41l4.59 4.59c.39.39 1.02.39 1.41 0l4.59-4.59a.996.996 0 0 0 0-1.41c-.39-.38-1.03-.39-1.42 0z"
                fill="#000"
                fill-opacity=".54"
              />
            </svg>
          </div>
          <div
            class="c22"
          >
            <ul
              class="c23"
              style="transform: translateY(calc(-2.2rem - 2.4rem));"
            >
              <li
                class="c24"
              >
                AM
              </li>
              <li
                class="c25"
              >
                PM
              </li>
            </ul>
          </div>
        </div>
        <div
          class="c26"
        >
          <button
            class="c27"
            type="button"
          >
            <span
              class="c15 c28"
            >
              Cancel
            </span>
          </button>
          <button
            class="c29"
            type="button"
          >
            <span
              class="c15 c28"
            >
              Apply
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
